draft-ietf-cdni-ci-triggers-rfc8007bis.xml

```diff
@@ -2028,1226 +1972,1147 @@ uCDN-A                         dCDN-B                         dCDN-C
-      <c>Object list in text format</c>
-      <c>RFCthis</c>
-      <c><xref target="object-list-type-text">Text</xref></c>
-        </texttable>
-   <t>[RFC Editor: Please replace RFCthis with the published RFC number for this document.]</t>
-   </section>
-   <section title="JSON Serialized Object List" anchor="object-list-type-json">
    <t>
-      This ObjectList type specifies a collection of objects encoded in JSON format, where each entry is encoded as an <xref target="object-entry"> ObjectEntry</xref> object. The entries in the JSON object list MAY have an object list type specified, allowing for a recursive object list structure.
+   To allow uCDN to check the status of multiple triggers in a single request, dCDN MAY maintain optional representations of 
+   the trigger collection, that contain a subset of all triggers, filtered based on a parameter
+   These filtered collection representations are "optional-to-implement", but if they
+   are implemented, dCDN MUST include links to them in the trigger collection.
    </t>
-   </section>
 
-   <section title="Text Object List" anchor="object-list-type-text">
-        <t>
-                   Unlike the <xref target="object-list-type-json">JSON Serialized Object List</xref>,
-                   the text-based object list will not support a recursive object list structure and 
-                   every object specified in it SHOULD be downloaded without additional processing.
-        </t>
-   </section>
 
-   <section title="ObjectEntry" anchor="object-entry">
    <t>
-      ObjectEntry is a metadata object describing an object and its associated metadata, to be used in JSON-encoded <xref target="object-list">ObjectList</xref> objects.
+   dCDN MAY provide filtered collections per <xref target="trigger-state">trigger state</xref>.
+   If implemented, dCDN SHOULD provide a separate filtered collection for every trigger state, and
+   advertise these collections in the trigger collection.
    </t>
    <t>
-        The following is an example of JSON serialized ObjectEntry objects:
-        <sourcecode type="json" name="objectentry-json"><![CDATA[
-[
-  { 
-    "href": "https://example.com/hls/35cdc008/index.m3u8", 
-    "type": "hls"
-  },
-  { 
-    "href": "https://example.com/dash/35cdc008/main.mpd", 
-    "type": "dash"
-  },
-  { 
-    "href": "https://example.com/img/35cdc008/thumb-l.jpg",
-    "size": 102600
-  },
-  { 
-    "href": "https://example.com/img/35cdc008/thumb-s.jpg", 
-    "size": 14535
-  },
-]
-]]></sourcecode>
+   dCDN MAY provide filtered collections per trigger label, as specified by uCDN when creating 
+   <xref target="ci-trigger-resource">a trigger resource</xref>.
+   If implemented, dCDN SHOULD provide a separate filtered collection for every unique trigger label, 
+   and advertise these collections in the collection of all Trigger resources.
+   </t>
+   <t>
+   Trigger Collections MUST use a MIME media type of "application/cdni; ptype=ci-trigger-collection".
```

Comment:
One thing I see that is potentially problematic is that v2 has changed the schema of the trigger collection.  coll-(active|pending|complete|failed) have been replaced with coll-state and coll-label, but the MIME type remains the same.  In RFC8007, the previous forms were listed as mandatory in the collection of all triggers, and now they are gone from the new spec.  Given that this is a breaking change for the schema, shouldn't we create a "v2" version of the MIME type for trigger collections?

Reply:
Agreed. Added intro language about new objects in the intro, replaced MIME type in the JSON snippets and added a new section in the IANA chapter, please review.